tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+    assert result_none is None
+    result = entry.check_retrieve_results_error((3, 3))
+    assert result is None
+    # I think I would need an actual transformer_retrieve argument in order to fully test this
+
+
+def test_entrypoint_check_metadata_needed():
+    """Tests entrypoint's check_metadata_needed function by looking at the agpypeline's
+     configuration.Configuration(), which has the attribute for metadata commented out"""
+    entry = entrypoint.__internal__()
+    assert entry.check_metadata_needed(configuration.Configuration()) is False
+
+
+def test_entrypoint_load_metadata_files():
+    """Tests entrypoint's load_metadata_files function by loading both an empty file and a metadata file
+    contained in the data folder and asserting that the respective results to the function call are the same
+    as these"""
+    entry = entrypoint.__internal__()
+    assert entry.load_metadata_files([]) == {'metadata': []}
+    orig = json.load(open("data/entrypoint_load_metadata_files.json"))
+    assert entry.load_metadata_files(["data/08f445ef-b8f9-421a-acf1-8b8c206c1bb8_metadata_cleaned.json"]) == orig
+
+
+def test_entrypoint_handle_check_continue_parse_continue_result():
+    """Tests entrypoint's handle_check_continue and parse_continue_result functions by passing in empty class
+     instances"""
+    entry = entrypoint.__internal__()
+    result = entry.handle_check_continue(algorithm.Algorithm(), environment.Environment(configuration.Configuration()),
+                                         {})
+    assert result == {}
+
+
+def test_entrypoint_handle_retrieve_files():
+    """Tests entrypoint's handle_retrieve_files function by passing in an empty class instances"""
+    entry = entrypoint.__internal__()
+    result = entry.handle_retrieve_files(environment.Environment(configuration.Configuration()), argparse.Namespace(),
+                                         [])
+    assert result is None
+
+
+def test_entrypoint_perform_processing():
+    """Tests entrypoint's perform_processing function by passing in empty class instances"""
+    namespace = argparse.Namespace()
+    namespace.file_list = []
+    namespace.working_space = []
+    entry = entrypoint.__internal__()
+    try:
+        entry.perform_processing(environment.Environment(configuration.Configuration()), algorithm.Algorithm(),
+                                 namespace, [])
+        assert True is False
+    except RuntimeError as error:
+        assert str(error) == "The Algorithm class method perform_process() must be overridden by a derived class"
+
+
+def test_entrypoint_handle_result():
+    """Tests entrypoint's handle_result function both by passing in empty parameters and by passing in a file path
+    to check for a result"""
+    entry = entrypoint.__internal__()
+    assert entry.handle_result({}) == {}
+    assert entry.handle_result({}, "", "") == {}
+    res = entry.handle_result({}, 'file', 'data/result.txt')
```

Comment:
Should check for and remove a data/result.txt file before making this call to ensure there isn't something hanging around